### PR TITLE
Controllers tests, DI, Test Helpers

### PR DESCRIPTION
**Pull request by zinkoviktor**

### CODE REVIEW

File: NetCoreRestApi/tests/UnitTests/Helpers/BaseComparer.cs [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+        public abstract bool AreEquals(T dto1, T dto2);
+
+        public virtual bool AreEquals(List<T> list1, List<T> list2)
+        {
+            if (list1.Count != list2.Count)
+            {
+                return false;
+            }
+
+            for (var i = 0; i < list1.Count; i++)
```
Comment (mdemii) [inline NetCoreRestApi/tests/UnitTests/Helpers/BaseComparer.cs]: rename i into index. Read code convention

File: NetCoreRestApi/tests/UnitTests/Helpers/CategoryHelpers/CategoryDtoHelper.cs [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+        {
+        }
+
+        public override bool AreEquals(CategoryDto model1, CategoryDto model2)
+        {
+            return Instance.AreObjectTypesEquals(model1, model2)
+                && model1.Id == model2.Id
+                    && model1.Name == model2.Name
+                        && model1.Description == model2.Description;
+        }
```
Comment (mdemii) [inline NetCoreRestApi/tests/UnitTests/Helpers/CategoryHelpers/CategoryDtoHelper.cs]: vertical align under Istance

File: NetCoreRestApi/tests/UnitTests/Helpers/CategoryHelpers/CategoryDtoHelper.cs [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+
+        public static List<CategoryDto> GetCategoryDtos()
```
Comment (mdemii) [inline NetCoreRestApi/tests/UnitTests/Helpers/CategoryHelpers/CategoryDtoHelper.cs]: Move this to each test

File: NetCoreRestApi/tests/UnitTests/Helpers/CategoryHelpers/CategoryModelHelper.cs [hunk 4]
```diff
@@ -0,0 +1,49 @@
+﻿using DataLayer.Models;
+using System.Collections.Generic;
+
+namespace UnitTests.Helpers.CategoryHelpers
+{
+    public class CategoryModelHelper : BaseComparer<CategoryModel>
```
Comment (mdemii) [inline NetCoreRestApi/tests/UnitTests/Helpers/CategoryHelpers/CategoryModelHelper.cs]: Rename helper into comparer

File: NetCoreRestApi/tests/UnitTests/Helpers/ObjectPropertiesComparer.cs [hunk 5]
```diff
@@ -0,0 +1,58 @@
+﻿using System.Collections.Generic;
+
+namespace UnitTests.Helpers
+{
+    public static class ObjectPropertiesComparer
```
Comment (mdemii) [inline NetCoreRestApi/tests/UnitTests/Helpers/ObjectPropertiesComparer.cs]: Remove this class

File: NetCoreRestApi/tests/UnitTests/Helpers/Tests/PerformanceTests.cs [hunk 6]
```diff
@@ -0,0 +1,42 @@
+﻿using DataLayer.Models;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+using System.Collections.Generic;
+using UnitTests.Helpers.CategoryHelpers;
+
+namespace UnitTests.Helpers.Tests
+{
+    [TestClass]
+    public class PerformanceTests
```
Comment (mdemii) [inline NetCoreRestApi/tests/UnitTests/Helpers/Tests/PerformanceTests.cs]: Remove

File: NetCoreRestApi/tests/UnitTests/WebApi/Controllers/ProductsControllerTests.cs [hunk 7]
```diff
@@ -1,101 +1,73 @@
...
+            InjectService<IUnitOfWorkContext>(mock.Object);
+
+            ConfigureServices();
+
+            Manager = ServiceProvider.GetRequiredService<IProductManager>();
+            Converter = ServiceProvider.GetRequiredService<IConverter<ProductDto, ProductModel>>();
         }
 
         [TestMethod]
         public void ProductsController_Get_ReturnsCategoryDtosList()
```
Comment (mdemii) [inline NetCoreRestApi/tests/UnitTests/WebApi/Controllers/ProductsControllerTests.cs]: Rename ProductsController_Get_ReturnsCategoryDtosList into Get_Action_Test

**Review by mdemii — CHANGES_REQUESTED** [review]
Fix comments

File: NetCoreRestApi/tests/UnitTests/Helpers/BaseComparer.cs (hunk 1 above)
Comment (zinkoviktor) [reply NetCoreRestApi/tests/UnitTests/Helpers/BaseComparer.cs]: fixed

File: NetCoreRestApi/tests/UnitTests/Helpers/CategoryHelpers/CategoryDtoHelper.cs (hunk 2 above)
Comment (zinkoviktor) [reply NetCoreRestApi/tests/UnitTests/Helpers/CategoryHelpers/CategoryDtoHelper.cs]: fixed

File: NetCoreRestApi/tests/UnitTests/Helpers/CategoryHelpers/CategoryDtoHelper.cs (hunk 3 above)
Comment (zinkoviktor) [reply NetCoreRestApi/tests/UnitTests/Helpers/CategoryHelpers/CategoryDtoHelper.cs]: fixed

File: NetCoreRestApi/tests/UnitTests/Helpers/CategoryHelpers/CategoryModelHelper.cs (hunk 4 above)
Comment (zinkoviktor) [reply NetCoreRestApi/tests/UnitTests/Helpers/CategoryHelpers/CategoryModelHelper.cs]: fixed

File: NetCoreRestApi/tests/UnitTests/Helpers/ObjectPropertiesComparer.cs (hunk 5 above)
Comment (zinkoviktor) [reply NetCoreRestApi/tests/UnitTests/Helpers/ObjectPropertiesComparer.cs]: fixed

File: NetCoreRestApi/tests/UnitTests/Helpers/Tests/PerformanceTests.cs (hunk 6 above)
Comment (zinkoviktor) [reply NetCoreRestApi/tests/UnitTests/Helpers/Tests/PerformanceTests.cs]: removed

File: NetCoreRestApi/tests/UnitTests/WebApi/Controllers/ProductsControllerTests.cs (hunk 7 above)
Comment (zinkoviktor) [reply NetCoreRestApi/tests/UnitTests/WebApi/Controllers/ProductsControllerTests.cs]: renamed

**Comment by zinkoviktor** [comment]
All comments resolved

**Pull request merged**